lfucg-exactions/client/js/actions/apiActions.js

```diff
@@ -1482,11 +1265,70 @@ export function getPagination(page) {
     return {
         type: API_CALL,
         endpoint: GET_PAGINATION,
-        url: () => {
-            if (!page.includes('paginatePage=true')) {
-                return `${page}?paginatePage=true`;
+        url: (getState) => {
+            const {
+                activeForm,
+            } = getState();
+            const {
+                currentPage,
+                page_size,
+            } = activeForm;
+            
+            if (!page) {
+                if (currentPage === '/account-ledger/') {
+                    return '/ledger/?paginatePage';
+                }
+
+                if (currentPage === '/project-cost/') {
+                    return '/estimate/?paginatePage';
+                }
+
+                if (page_size) {
+                    return `${currentPage}?paginatePage&pageSize=${page_size}`;
+                }
+                return `${currentPage}?paginatePage`;
+            }
+            if (!page.includes('paginatePage')) {
+                return `${page}?paginatePage`;
             }
             return `${page}`;
         },
     };
 }
+
+export function searchQuery() {
+    return {
+        type: API_CALL,
+        endpoint: SEARCH_QUERY,
+        url: (getState) => {
+            const {
+                activeForm,
+            } = getState();
+
+
+            let query_all = `${activeForm.currentPage}?paginatePage`;
+
+            if (activeForm.currentPage === '/account-ledger/') {
+                query_all = '/ledger/?paginatePage';
+            }
+
+            if (activeForm.currentPage === '/project-cost/') {
+                query_all = '/estimate/?paginatePage';
+            }
+
+            map((a) => {
```

Comment:
@JMStewart00 We aren't really using `map` properly here. Map should return data with same number of entries where each entry has been modified some. What you are really doing is a for each loop to mutate the `query_all` variable.

This isn't a big deal, but if you have a few minutes, I would be happy to show you how to do this in a filter/map/reduce functional style.